yearn/contracts/SaddleStrategy.sol

```diff
@@ -0,0 +1,395 @@
+        );
+        //TODO: Check if both the LP Token and KEEP token should be migrated to
+        // the new strategy, or just the LP token.
+    }
+
+    /// @notice This method is defined in the BaseStrategy contract and is meant
+    ///         to perform any strategy unwinding or other calls necessary to
+    ///         capture the free return this strategy has generated since the
+    ///         last time its core position(s) were adjusted. Examples include
+    ///         unwrapping extra rewards. This call is only used during normal
+    ///         operation of a strategy, and should be optimized to minimize
+    ///         losses as much as possible.
+    /// @param debtOutstanding Will be 0 if the strategy is not past the
+    ///        configured debt limit, otherwise its value will be how far past
+    ///        the debt limit the strategy is. The strategy's debt limit is
+    ///        configured in the vault.
+    /// @return profit Amount of realized profit.
+    /// @return loss Amount of realized loss.
+    /// @return debtPayment Amount of repaid debt. The value of debtPayment
+    ///         should be less than or equal to debtOutstanding. It is okay for
+    ///         it to be less than debtOutstanding, as that should only used as
+    ///         a guide for how much is left to pay back. Payments should be
+    ///         made to minimize loss from slippage, debt, withdrawal fees, etc.
+    function prepareReturn(uint256 debtOutstanding)
+        internal
+        override
+        returns (
+            uint256 profit,
+            uint256 loss,
+            uint256 debtPayment
+        )
+    {
+        /* solhint-disable not-rely-on-time */
+        // Get the initial balance of the vault's underlying token under
+        // strategy management.
+        uint256 initialWantBalance = balanceOfWant();
+
+        // Get KEEP rewards from the Saddle reward pool.
+        ILPRewardsTBTCV2Saddle(lpRewardsTbtcV2Saddle).getReward();
+
+        // Buy WBTC using obtained KEEP tokens.
+        uint256 keepBalance = IERC20(keep).balanceOf(address(this));
+        if (keepBalance > 0) {
+            IERC20(keep).safeIncreaseAllowance(uniswap, keepBalance);
+
+            address[] memory path = new address[](3);
+            path[0] = keep;
+            path[1] = weth;
+            path[2] = wbtc;
+
+            IUniswapV2Router(uniswap).swapExactTokensForTokens(
+                keepBalance,
+                0,
+                path,
+                address(this),
+                now
+            );
+        }
+
+        // Deposit acquired WBTC to the Saddle pool to gain additional
+        // vault's underlying tokens.
+        uint256 wbtcBalance = IERC20(wbtc).balanceOf(address(this));
+        if (wbtcBalance > 0) {
+            IERC20(wbtc).safeIncreaseAllowance(swap, wbtcBalance);
+
+            uint256[] memory amounts = new uint256[](4);
+            amounts[1] = wbtcBalance;
```

Comment:
Can we replace it with a direct function parameter like:
```
[0, wbtcBalance, 0, 0]
```
passed to the `addLiquidity`? This way it looks more concise.

Reply:
Unfortunately not:
`[0, wbtcBalance, 0, 0]` has type `uint256[4] memory`, but the function needs `uint256[] memory`.
It can only be done with `new`.

Reply:
Ok, got it!